service/matching/taskWriter.go

```diff
@@ -179,41 +198,37 @@ writerLoop:
 	for {
 		select {
 		case request := <-w.appendCh:
-			{
-				// read a batch of requests from the channel
-				reqs := []*writeTaskRequest{request}
-				reqs = w.getWriteBatch(reqs)
-				batchSize := len(reqs)
-
-				maxReadLevel := int64(0)
-
-				taskIDs, err := w.allocTaskIDs(batchSize)
-				if err != nil {
-					w.sendWriteResponse(reqs, nil, err)
-					continue writerLoop
-				}
-
-				var tasks []*persistencespb.AllocatedTaskInfo
-				for i, req := range reqs {
-					tasks = append(tasks, &persistencespb.AllocatedTaskInfo{
-						TaskId: taskIDs[i],
-						Data:   req.taskInfo,
-					})
-					maxReadLevel = taskIDs[i]
-				}
-
-				resp, err := w.appendTasks(tasks)
-				w.sendWriteResponse(reqs, resp, err)
-				// Update the maxReadLevel after the writes are completed.
-				if maxReadLevel > 0 {
-					atomic.StoreInt64(&w.maxReadLevel, maxReadLevel)
-				}
+			// read a batch of requests from the channel
+			reqs := []*writeTaskRequest{request}
+			reqs = w.getWriteBatch(reqs)
```

Comment:
nit: getWriteBatch should not take input. Code could looks like:
reqs := w.getWriteBatch()
reqs = append(reqs, request)

Reply:
we can restructure the code later for maintainability, this PR only aims to provide at most once start / stop

Reply:
sg